component/prometheus/operator/common/crdmanager.go

```diff
@@ -12,10 +12,13 @@ import (
 	"github.com/go-kit/log/level"
 	"github.com/grafana/agent/component"
 	"github.com/grafana/agent/component/prometheus"
+	"github.com/grafana/agent/pkg/cluster"
+	"github.com/prometheus/common/model"
 	"github.com/prometheus/prometheus/config"
 	"github.com/prometheus/prometheus/discovery"
 	"github.com/prometheus/prometheus/discovery/targetgroup"
 	"github.com/prometheus/prometheus/scrape"
+	"github.com/rfratto/ckit/shard"
```

Comment:
This is now available on grafana/ckit

```suggestion
	"github.com/grafana/ckit/shard"
```